plugins/base/src/main/java/edu/wpi/first/shuffleboard/plugin/base/widget/FieldWidget.java

```diff
@@ -0,0 +1,273 @@
+package edu.wpi.first.shuffleboard.plugin.base.widget;
+
+import com.google.common.collect.ImmutableList;
+import com.google.gson.Gson;
+import edu.wpi.first.shuffleboard.api.prefs.Group;
+import edu.wpi.first.shuffleboard.api.prefs.Setting;
+import edu.wpi.first.shuffleboard.api.widget.Description;
+import edu.wpi.first.shuffleboard.api.widget.ParametrizedController;
+import edu.wpi.first.shuffleboard.api.widget.SimpleAnnotatedWidget;
+import edu.wpi.first.shuffleboard.plugin.base.data.FieldData;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.IOException;
+import java.io.Reader;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javafx.beans.property.Property;
+import javafx.beans.property.SimpleObjectProperty;
+import javafx.fxml.FXML;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.BorderPane;
+import javafx.scene.layout.Pane;
+import javafx.scene.paint.Color;
+import javafx.scene.paint.Paint;
+import javafx.scene.shape.Circle;
+
+@Description(name = "Field", dataTypes = FieldData.class)
+@SuppressWarnings("EmptyCatchBlock")
+@ParametrizedController("FieldWidget.fxml")
+public class FieldWidget extends SimpleAnnotatedWidget<FieldData> {
+  @FXML
+  private Pane root;
+
+  @FXML
+  private BorderPane pane;
+
+  @FXML
+  private ImageView backgroundImage;
+
+  @FXML
+  private ImageView robot;
+
+  private double imageStartX;
+  private double imageStartY;
+  private double imageEndX;
+  private double imageEndY;
+  private double fieldWidth;
+  private double fieldHeight;
+
+  private enum Game {
+    A2018_Power_Up,
+    A2019_Deep_Space,
+    A2020_Infinite_Recharge,
+    A2021_Barrel_Racing_Path,
+    A2021_Bounce_Path,
+    A2021_Galactic_Search_A,
+    A2021_Galactic_Search_B,
+    A2021_Infinite_Recharge,
+    A2021_Slalom_Path;
+
+    public String json() {
+      return "field/" + this.name().substring(1).toLowerCase().replaceFirst("_", "-").replaceAll("_", "") + ".json";
+    }
+
+    @Override
+    public String toString() {
+      return this.name().substring(1).replaceAll("_", " ");
+    }
+  }
+
+  private final Property<Game> game =
+          new SimpleObjectProperty<>(Game.A2021_Infinite_Recharge);
+
+  private final Map<String, Paint> colors = new HashMap<>();
+
+  private static final double ROBOT_SIZE = 30;
+
+  @FXML
+  private void initialize() {
+    robot.setImage(
+            new Image(getClass().getResource("field/robot.png").toExternalForm()));
+    robot.setFitWidth(ROBOT_SIZE);
+    robot.setFitHeight(ROBOT_SIZE);
+    setGame(game.getValue());
+    game.addListener(__ -> {
+      setGame(game.getValue());
+      centerImage();
+      updateRobotPosition();
```

Comment:
Is the coordinate system based on the image and root height/width? Alternatively, does this need to be called here?

Reply:
The circle placements are dependent on the field width and height as well as the image fitting and field startX, endX, startY and endY, so if this wasn't called it wouldn't be placed correctly